lizard_ext/lizardfans.py

```diff
@@ -27,28 +32,35 @@ def __call__(tokens, reader):
         So don't store any data with an extension object.
         """
         reader.context.fileinfo.fan_in = reader.context.fileinfo.fan_out = 0
-        for token in tokens:
-            yield token
+        return tokens
 
     def fans(self, fileinfo):
         """
         Preparation of calculating the fan in and fan out (prototype)
         """
+        infile = Infile()
         self.name_list = [func.name for func in fileinfo.function_list]
         for func in fileinfo.function_list:
-            body = self.method_open(func)
+            if not func.filename == infile.name:
+                infile = self.open_method(func.filename)
+            body = self.store_method(infile, func)
```

Comment:
You may try to make self.open_method(func.filename) into the __init__ of the InFile class and move the static methods to the InFile class.